api/controllers/login.js

```diff
@@ -15,55 +15,50 @@ const get = (request, response) => {
 }
 
 const post = (request, response) => {
-  const serviceTicket = request.body.ticket
+  const accessToken = request.body.access_token
 
-  if (!serviceTicket) {
+  if (!accessToken) {
     util.buildUnauthorizedResponse(response)
     return
   }
 
-  validateTicket(serviceTicket, (data) => {
-    if (!data) {
-      util.buildInternalErrorResponse(response)
+  lookupProfile(accessToken).then(({ profile, error }) => {
+    if (error?.unauthorized) {
+      logger.debug('Login failure: unauthorized')
+      util.buildUnauthorizedResponse(response)
       return
     }
 
-    const json = JSON.parse(data)
-    if (json.serviceResponse.authenticationFailure) {
-      logger.debug('Login failure: ' + json.serviceResponse.authenticationFailure.description)
-      util.buildUnauthorizedResponse(response)
+    const guid = profile?.ssoguid
+    if (!guid) {
+      util.buildInternalErrorResponse(response)
       return
     }
-    const guid = json.serviceResponse.authenticationSuccess.attributes.ssoGuid[0].toLowerCase()
     buildJwt(guid, response)
+  }).catch(() => {
+    util.buildInternalErrorResponse(response)
   })
 }
 
-const validateTicket = (serviceTicket, callback) => {
-  const path = '/cas/p3/serviceValidate'
-  const service = 'service=' + encodeURIComponent(process.env.THE_KEY_SERVICE_URL)
-  const ticket = 'ticket=' + encodeURIComponent(serviceTicket)
-  const format = 'format=JSON'
-  const options = {
-    hostname: 'thekey.me',
-    path: path + '?' + service + '&' + ticket + '&' + format,
-    method: 'GET'
-  }
-
-  const request = https.request(options, (response) => {
-    response.setEncoding('utf8')
-
-    response.on('data', (data) => {
-      callback(data)
-    })
-  })
-
-  request.on('error', (error) => {
-    logger.error(error)
-    callback(null)
+async function lookupProfile (accessToken) {
+  const res = await fetch(`${process.env.OKTA_ISSUER}/v1/userinfo`, {
```

Comment:
Ah, I see. You're doing what I had suggested in the other spot here. You can disregard the other comments.